shpkpr/marathon/application.py

```diff
@@ -0,0 +1,74 @@
+# stdlib imports
+import json
+import os
+
+# third-party imports
+import jsonschema
+from cached_property import cached_property
+
+# local imports
+from shpkpr import exceptions
+
+
+def default_schema():
+    """Path to the default schema used to validate application definitions
+    before sending to marathon.
+    """
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), "schema", "app.json")
+
+
+class ValidationError(exceptions.ShpkprException):
+    exit_code = 2
+
+    def format_message(self):
+        return 'Unable to validate application: %s' % self.message
+
+
+class ApplicationError(exceptions.ShpkprException):
+    exit_code = 2
+
+
+class MarathonApplication(object):
```

Comment:
There might be another option as well, since it's got a JSON Schema there might be a Python library to generate classes from that schema, or objects from it. It potentially gives you option 1 without having to write a lot of boilerplate code.

Either way, a step back and think on it sounds good. If you want we can pair on it a bit tomorrow and see what works well.